irc/varys/varys.go

```diff
@@ -160,6 +164,29 @@ func (v *Varys) SendRaw(params SendRawParams, _ *struct{}) error {
 	return nil
 }
 
+type GetNickParams struct {
+	Nick string
+}
+
+func (v *Varys) GetNick(uid string, nick *GetNickParams) error {
+	if conn, ok := v.uidToConns[uid]; ok {
+		nick.Nick = conn.GetNick()
+	}
+	return nil
+}
```

Comment:
See other comments (net/rpc docs link + refer to GetUIDToNicks implmeentation).

```suggestion
func (v *Varys) GetNick(uid string, result *string) error {
	if conn, ok := v.uidToConns[uid]; ok {
		*result = conn.GetNick()
	}
	return nil
}
```

The key point here is that the result can be any pointer type, so we don't need a struct.

Same thing applies to Connected.

Reply:
I originally did something along the lines of any pointer type and got linter complaints about ineffectual assignments but I did do it a bit differently. I'm away from computer and typing this at the phone but will try your suggested changes when I'm back at the computer.